assets/js/modules/thank-with-google/components/common/PostTypesSelect.js

```diff
@@ -52,21 +56,30 @@ export default function PostTypesSelect() {
 				return;
 			}
 
+			let newCTAPostTypes;
+
 			if ( chipID === 'all' ) {
 				if ( ctaPostTypes?.length === postTypes?.length ) {
-					setCTAPostTypes( [] );
+					newCTAPostTypes = [];
 				} else {
-					setCTAPostTypes( postTypes.map( ( { slug } ) => slug ) );
+					newCTAPostTypes = postTypes.map( ( { slug } ) => slug );
 				}
 			} else if ( ctaPostTypes?.includes( chipID ) ) {
-				setCTAPostTypes(
-					ctaPostTypes.filter( ( postType ) => postType !== chipID )
+				newCTAPostTypes = ctaPostTypes.filter(
+					( postType ) => postType !== chipID
 				);
 			} else {
-				setCTAPostTypes( [ ...( ctaPostTypes || [] ), chipID ] );
+				newCTAPostTypes = [ ...( ctaPostTypes || [] ), chipID ];
 			}
+
+			setCTAPostTypes( newCTAPostTypes );
+			trackEvent(
+				`${ viewContext }_thank-with-google`,
+				'change_cta_post_types',
+				newCTAPostTypes.toString()
```

Comment:
Although `toString` works, I'd suggest calling `join` instead - or even `join( ',' )` to be explicit, as it's a bit more descriptive in terms of the intent here.